pymedphys/_gui/streamlit/mudensity-compare.py

```diff
@@ -814,33 +818,46 @@ def trf_input_method(patient_id="", key_namespace="", **_):
 
     headers
 
+    deliveries = cached_deliveries_loading(tables, delivery_from_trf)
+
+    individual_identifiers = [
+        f"{path.parent.parent.parent.parent.name} {path.parent.name}"
+        for path in selected_filepaths
+    ]
+
+    identifier = f"TRF ({individual_identifiers[0]})"
+
     """
     #### Corresponding Mosaiq SQL Details
     """
 
-    mosaiq_details = get_logfile_mosaiq_info(headers)
-    mosaiq_details = mosaiq_details.drop("beam_completed", axis=1)
-
-    mosaiq_details
-
-    patient_names = set()
-    for _, row in mosaiq_details.iterrows():
-        row
-        patient_name = utl_patient.convert_patient_name_from_split(
-            row["last_name"], row["first_name"]
+    try:
+        mosaiq_details = get_logfile_mosaiq_info(headers)
+        use_mosaiq = True
+    except KeyError:
+        use_mosaiq = False
+        st.write(
+            NoMosaiqAccess(
+                "Need Mosaiq access to determine patient name. "
+                "Patient name set to 'Unknown'."
+            )
         )
-        patient_names.add(patient_name)
+        patient_name = "Unknown"
```

Comment:
I would set this above the GUI part, and then pass in patient_name to the GUI.
Could also put the identifier that is known in to the GUI so it's clear what is happening to which data.

Reply:
A little unsure what you mean here. I'll think on it more when I'm at a computer.

Reply:
```
        use_mosaiq = False
        patient_name = "Unknown"
        st.write(
            NoMosaiqAccess(
                "Need Mosaiq access to determine patient name. "
                f"Patient name set to '{patient_name}'."
            )
        )
```

Reply:
In practice, the user also provides a reference set of data (Monaco, Mosaiq, or RT plan DICOM). The GUI happily extracts the patient name from the reference and having the evaluation dataset be missing the name at the end of the day has no impact.

Can we revisit this one after this PR?